src/PhpSpec/Locator/PSR0/PSR0Locator.php

```diff
@@ -357,7 +357,7 @@ private function createResourceFromSpecFile($path)
      */
     private function validatePsr0Classname($classname)
     {
-        $pattern = '/^([a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*[\/\\\\]?)*[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/';
+        $pattern = '/^[^\r\n]([a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*[\/\\\\]?)*[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/';
```

Comment:
This isn't correct, you're allowing any char that isn't `\r` or `\n` to be the first character of the class name, including a bunch of invalid ones. I\'m not really clear why this makes your test pass, to be honest.